src/index.html

```diff
@@ -7,6 +7,102 @@
     <title>Vite App</title>
   </head>
   <body>
+    <nav
+      class="bg-slate-100 p-10"
+      x-data="nav"
+      @keyup.escape.window="popStack(1)"
+    >
+      <ul class="inline-flex gap-6" @click.outside="resetStack()">
```

Comment:
@click.outside isn't 100% ideal on this tag, but does an ok job. 

- The `<ul>` has to be inline flex, so it only extends the with of the nav items. 
- Clickin in the gap between nav items doesn't close the navigation (because we technically didn't click outside of the ul).